### PR TITLE
Georestore command for MySQL and PostgreSQL CLI.

### PR DESCRIPTION
**Pull request by rohit-joy**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `./scripts/ci/test_static.sh` locally? (`pip install pylint flake8` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli-extensions/blob/master/docs/extension_summary_guidelines.md).

### CODE REVIEW

File: src/rdbms/azext_rdbms/_help.py [hunk 1]
```diff
@@ -31,153 +22,17 @@ def add_helps(command_group, server_type):
...
-                    - name: Restore 'testsvr2' to 'testsvrnew', where 'testsvrnew' is in a different resource group than the backup.
+                    - name: Georestore 'testsvr' as 'testsvrnew'.
+                      text: az {0} server georestore -g testgroup -n testsvrnew --source-server testsvr -l westus2"
+                    - name: Georestore 'testsvr2' to 'testsvrnew', where 'testsvrnew' is in the same resource group as the original server but in a different location.
                       text: |
-                        az {0} server restore -g testgroup -n testsvrnew \\
-                            -s "/subscriptions/${{SubID}}/resourceGroups/${{ResourceGroup}}/providers/Microsoft.DBfor{1}/servers/testsvr2" \\
-                            --restore-point-in-time "2017-06-15T13:10:00Z"
+                        az {0} server georestore -g testgroup -n testsvrnew \\
+                            -s "/subscriptions/${{SubID}}/resourceGroups/${{ResourceGroup}}/providers/Microsoft.DBfor{1}/servers/testsvr2" -l westus2 --sku-name GP_Gen5_2 "
```
Comment (derekbekoe) [inline src/rdbms/azext_rdbms/_help.py]: There is an extra `"` at the end of the line that looks like should be removed.

File: src/rdbms/azext_rdbms/_help.py [hunk 2]
```diff
@@ -31,153 +22,17 @@ def add_helps(command_group, server_type):
...
+    helps['{} server georestore'.format(command_group)] = """
                 type: command
-                short-summary: Restore a server from backup.
+                short-summary: Georestore a server from backup.
                 examples:
-                    - name: Restore 'testsvr' as 'testsvrnew'.
-                      text: az {0} server restore -g testgroup -n testsvrnew --source-server testsvr --restore-point-in-time "2017-06-15T13:10:00Z"
-                    - name: Restore 'testsvr2' to 'testsvrnew', where 'testsvrnew' is in a different resource group than the backup.
+                    - name: Georestore 'testsvr' as 'testsvrnew'.
+                      text: az {0} server georestore -g testgroup -n testsvrnew --source-server testsvr -l westus2"
```
Comment (derekbekoe) [inline src/rdbms/azext_rdbms/_help.py]: There is an extra `"` at the end of the line that looks like should be removed.

File: src/rdbms/azext_rdbms/commands.py [hunk 3]
```diff
@@ -35,124 +25,10 @@ def load_command_table(self, _):
...
-        g.command('delete', 'delete', confirmation=True)
-        g.command('show', 'get')
-        g.custom_command('list', '_server_list_custom_func')
-        g.generic_update_command('update',
-                                 getter_name='_server_update_get', getter_type=rdbms_custom,
-                                 setter_name='_server_update_set', setter_type=rdbms_custom, setter_arg_name='parameters',
-                                 custom_func_name='_server_update_custom_func')
-        g.generic_wait_command('wait', getter_name='_server_mysql_get', getter_type=rdbms_custom)
+        g.custom_command('create', '_server_create')
+        g.custom_command('georestore', '_server_georestore', no_wait_param='no_wait')
```
Comment (derekbekoe) [inline src/rdbms/azext_rdbms/commands.py]: To support no wait, we now support just doing `supports_no_wait=True`

See "(4) Supporting --no-wait" in https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#register-commands

File: src/rdbms/azext_rdbms/commands.py [hunk 4]
```diff
@@ -35,124 +25,10 @@ def load_command_table(self, _):
...
-        g.command('show', 'get')
-        g.command('list', 'list_by_server')
-
-    with self.command_group('postgres db', postgres_db_sdk) as g:
-        g.command('create', 'create_or_update')
-        g.command('delete', 'delete', confirmation=True)
-        g.command('show', 'get')
-        g.command('list', 'list_by_server')
+        g.custom_command('create', '_server_create')
+        g.custom_command('georestore', '_server_georestore', no_wait_param='no_wait')
```
Comment (derekbekoe) [inline src/rdbms/azext_rdbms/commands.py]: Same here.

File: src/rdbms/azext_rdbms/custom.py [hunk 5]
```diff
@@ -8,20 +8,75 @@
 from azure.cli.core.commands.client_factory import get_subscription_id
 from msrestazure.tools import resource_id, is_valid_resource_id, parse_resource_id  # pylint: disable=import-error
 from azext_rdbms.mysql.operations.servers_operations import ServersOperations
-from ._client_factory import get_mysql_management_client, get_postgresql_management_client
 
 SKU_TIER_MAP = {'Basic': 'b', 'GeneralPurpose': 'gp', 'MemoryOptimized': 'mo'}
 
 
-# Need to replace source server name with source server id, so customer server restore function
+def _server_create(cmd, client, resource_group_name, server_name, sku_name, no_wait=False,
```
Comment (derekbekoe) [inline src/rdbms/azext_rdbms/custom.py]: Does the method actually use `no_wait`?
Doesn't look like it from the diff.

File: src/rdbms/azext_rdbms/custom.py [hunk 6]
```diff
@@ -31,143 +86,14 @@ def _server_restore(cmd, client, resource_group_name, server_name, parameters, n
...
     try:
-        source_server_object = client.get(id_parts['resource_group'], id_parts['name'])
-        parameters.location = source_server_object.location
+        source_server_object = client.get(source_server_id_parts['resource_group'], source_server_id_parts['name'])
+        if parameters.sku.name is None:
+            parameters.sku.name = source_server_object.sku.name
     except Exception as e:
         raise ValueError('Unable to get source server: {}.'.format(str(e)))
 
     return client.create(resource_group_name, server_name, parameters, raw=no_wait)
```
Comment (derekbekoe) [inline src/rdbms/azext_rdbms/custom.py]: `raw=no_wait` no longer works for newer SDKs that are generated with Autorest 3.0. The polling will still happen which is not the intent here.

Instead
```
from azure.cli.core.util import sdk_no_wait
return sdk_no_wait(no_wait , client.create, resource_group_name, server_name, parameters)
```

You can see "(4) Supporting --no-wait" in https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#register-commands

File: src/rdbms/setup.py [hunk 7]
```diff
@@ -8,7 +8,7 @@
 from codecs import open
 from setuptools import setup, find_packages
 
-VERSION = "0.0.4"
+VERSION = "0.0.5"
```
Comment (derekbekoe) [inline src/rdbms/setup.py]: Change the description and long_description to "Support for Azure MySQL and Azure PostgreSQL." (or something else) to conform to the new guidelines on this - https://github.com/Azure/azure-cli-extensions/blob/master/docs/extension_summary_guidelines.md

Comment (derekbekoe) [reply src/rdbms/setup.py]: Also, mark the extension as preview which I think this is.
https://github.com/Azure/azure-cli/blob/dev/doc/extensions/metadata.md#azextispreview

File: src/rdbms/azext_rdbms/custom.py (hunk 5 above)
Comment (rohit-joy) [reply src/rdbms/azext_rdbms/custom.py]: It doesn't.

File: src/rdbms/azext_rdbms/_help.py (hunk 1 above)
Comment (rohit-joy) [reply src/rdbms/azext_rdbms/_help.py]: Removed

File: src/rdbms/azext_rdbms/_help.py [hunk 8]
```diff
@@ -31,153 +22,17 @@ def add_helps(command_group, server_type):
...
-    helps['{} server restore'.format(command_group)] = """
+    helps['{} server georestore'.format(command_group)] = """
                 type: command
-                short-summary: Restore a server from backup.
+                short-summary: Georestore a server from backup.
                 examples:
-                    - name: Restore 'testsvr' as 'testsvrnew'.
-                      text: az {0} server restore -g testgroup -n testsvrnew --source-server testsvr --restore-point-in-time "2017-06-15T13:10:00Z"
-                    - name: Restore 'testsvr2' to 'testsvrnew', where 'testsvrnew' is in a different resource group than the backup.
+                    - name: Georestore 'testsvr' as 'testsvrnew'.
```
Comment (ksundeepsatya) [inline src/rdbms/azext_rdbms/_help.py]: Georestore 'testsvr' as 'testsvrnew' where 'testsvrnew' is in same resource group as 'testsvr'

File: src/rdbms/azext_rdbms/_help.py [hunk 9]
```diff
@@ -31,153 +22,17 @@ def add_helps(command_group, server_type):
...
+                      text: az {0} server georestore -g testgroup -n testsvrnew --source-server testsvr -l westus2"
+                    - name: Georestore 'testsvr2' to 'testsvrnew', where 'testsvrnew' is in the same resource group as the original server but in a different location.
```
Comment (ksundeepsatya) [inline src/rdbms/azext_rdbms/_help.py]: Georestore 'testsvr2' to 'testsvrnew', where 'testsvrnew' is in different resource group than the original server. Location doesn't matter

File: src/rdbms/azext_rdbms/_params.py [hunk 10]
```diff
@@ -17,39 +15,31 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
...
-            c.argument('restore_point_in_time', help='The point in time to restore from (ISO8601 format), e.g., 2017-04-26T02:10:00+08:00')
-
-        with self.argument_context('{} server configuration set'.format(command_group)) as c:
-            c.argument('value', help='Value of the configuration. If not provided, configuration value will be set to default.', validator=configuration_value_validator)
-            c.ignore('source')
+        with self.argument_context('{} server georestore'. format(command_group)) as c:
+            c.argument('location', arg_type=get_location_type(self.cli_ctx), required=True)
+            c.argument('sku_name', options_list=['--sku-name'], required=False, help='The name of the sku, typically, tier + family + cores, e.g. B_Gen4_1, GP_Gen5_8.')
+            c.argument('source_server', options_list=['--source-server', '-s'], required=True, help='The name or ID of the source server to restore from.')
+            c.argument('backup_retention', options_list=['--backup-retention'], type=int, help='The max days of retention, unit is days.')
```
Comment (ksundeepsatya) [inline src/rdbms/azext_rdbms/_params.py]: Add geo-redundant-backup option

Comment (rohit-joy) [reply src/rdbms/azext_rdbms/_params.py]: Done

File: src/rdbms/azext_rdbms/_help.py (hunk 9 above)
Comment (rohit-joy) [reply src/rdbms/azext_rdbms/_help.py]: Updated.

**Pull request merged**